src/tests/ftest/util/pool_create_all_base.py

```diff
@@ -213,6 +233,7 @@ def check_pool_recycling(self, pool_count, scm_delta_bytes, nvme_delta_bytes=Non
             self.log.info(
                 "Pool %d created: scm_size=%d, nvme_size=%d", index, *pool_size)
             self.pool[index].destroy()
+            self.wait_for_pool_destroyed(pool_idx=index)
```

Comment:
I think it's cleaner to pass the pool object instead of the index. Then, `wait_for_pool_destroyed` doesn't have to know anything about `self.pool` being a list

Reply:
I realized that I didn't have to create `wait_for_pool_destroyed()`, so moved the code block here.
I forgot to check if my change affects other places. Ran `test_recycle_pools_hw` this time.